src/Microsoft.AspNet.Diagnostics/ErrorPageMiddleware.cs

```diff
@@ -109,7 +115,7 @@ private Task DisplayException(HttpContext context, Exception ex)
                         Function = string.Empty
                     };
 
-                    ReadFrameContent(frame, fileContent, item.StartLine, item.EndLine);
+                    GetErrorCodeBlock(frame, fileContent, item.StartLine, item.EndLine);
```

Comment:
Not familiar with how the `compilationFailure.Messages` are organized so can't say I know an exact reproduction of a multi-line error. I'd imagine C# block code which is spread across multiple lines would be the route to take however.